Makefile.dep

```diff
@@ -392,7 +392,14 @@ endif
 
 ifneq (,$(filter stdio_uart,$(USEMODULE)))
   USEMODULE += isrpipe
-  FEATURES_REQUIRED += periph_uart
+  ifneq (,$(filter newlib_syscalls_mips_uhi,$(USEMODULE)))
+    # The periph_uart driver is optional when using MIPS UHI syscalls (debug
+    # UART is used on PIC32 boards, but not on mips-malta)
+    FEATURES_OPTIONAL += periph_uart
```

Comment:
Now that I look at it on its own, does this somehow means that `newlib_syscalls_mips_uhi` implements the `periph_uart` functions on mips malta and could be described the other way around ? Like adding this somewhere in the `mips3r2_generic` cpu for mips-malta:

    ifneq (,$(filter newlib_syscalls_mips_uhi,$(USEMODULE))
      FEATURES_PROVIDED += periph_uart
    endif

With of course a comment saying it is not really an uart but implements the interface.

The fact that it adds ` periph_uart` pseudomodule to USEMODULE should not be a problem in that case

Reply:
From doing `mips-mti-elf-objdump -t examples/hello-world/bin/mips-malta/hello-world.elf  | grep stdio ` I get no reference to the `stdio_uart` module. I could even compile `hello-world` when removing `stdio_uart`.

Maybe a solution would be more to not include `stdio_uart` for `mips_malta`.

It compiles with:

``` diff
index 25cdb71af..e12d35924 100644
--- a/Makefile.dep
+++ b/Makefile.dep
@@ -367,7 +367,7 @@ ifneq (,$(filter newlib,$(USEMODULE)))
   ifeq (,$(filter newlib_syscalls_%,$(USEMODULE)))
     USEMODULE += newlib_syscalls_default
   endif
-  ifeq (,$(filter stdio_rtt,$(USEMODULE)))
+  ifeq (,$(filter stdio_%,$(USEMODULE)))
     USEMODULE += stdio_uart
   endif
 endif
diff --git a/cpu/mips32r2_generic/Makefile.include b/cpu/mips32r2_generic/Makefile.include
index efb2a88b1..27e261919 100644
--- a/cpu/mips32r2_generic/Makefile.include
+++ b/cpu/mips32r2_generic/Makefile.include
@@ -2,6 +2,14 @@ export MEMORY_BASE=0x80000000
 export MEMORY_SIZE=1M
 export APP_START=0x80000000
 
+
+# newlib_syscalls_mips_uhi provides directly the 'stdio' functions
+# This prevents using 'stdio_uart' module
+ifeq ($(USE_UHI_SYSCALLS),1)
+  USEMODULE += stdio_mips_uhi
+endif
+PSEUDOMODULES += stdio_mips_uhi
+
 include $(RIOTMAKE)/arch/mips.inc.mk
 include $(RIOTCPU)/mips32r2_common/Makefile.include
 
```

I did not test it though.

Reply:
With this, it builds all compatible applications.